internal/errors/transport.go

```diff
@@ -122,3 +122,15 @@ func (e ProcedureFailedError) Error() string {
 func (e ProcedureFailedError) AsHandlerError() HandlerError {
 	return HandlerUnexpectedError(e)
 }
+
+// UnknownRPCModeError is a failure to process a request because the RPC
+// mode is unkown to the transport
+//TODO: add tests for this error
+type UnknownRPCModeError struct {
```

Comment:
I'm also open to calling the concept `Message Pattern` - depends on the teams opinion. cc @bombela @kriskowal @abhinav @willhug